native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -0,0 +1,621 @@
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeFactory;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.NodeParser;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.RecordFieldNode;
+import io.ballerina.compiler.syntax.tree.RecordFieldWithDefaultValueNode;
+import io.ballerina.compiler.syntax.tree.RecordTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.persist.components.Class;
+import io.ballerina.persist.components.Function;
+import io.ballerina.persist.components.IfElse;
+import io.ballerina.persist.components.TypeDescriptor;
+import io.ballerina.persist.objects.Entity;
+import io.ballerina.persist.objects.FieldMetaData;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+
+/**
+ * Class to ballerina files as syntax trees.
+ */
+public class BalSyntaxTreeGenerator {
+
+    /**
+     * method to read ballerina files.
+     */
+    public static ArrayList<Entity> readBalFiles(Path filePath, String module) throws IOException {
+        ArrayList<Entity> entityArray = new ArrayList<>();
+        int index = -1;
+        ArrayList<String> keys = new ArrayList<>();
+        String tableName = null;
+        int count;
+        SyntaxTree balSyntaxTree = SyntaxTree.from(TextDocuments.from(Files.readString(filePath)));
+        ModulePartNode rootNote = balSyntaxTree.rootNode();
+        NodeList<ModuleMemberDeclarationNode> nodeList = rootNote.members();
+        for (ModuleMemberDeclarationNode moduleNode : nodeList) {
+            if (moduleNode.kind() != SyntaxKind.TYPE_DEFINITION) {
+                continue;
+            }
+            Collection<ChildNodeEntry> temp = moduleNode.childEntries();
+            for (ChildNodeEntry child : temp) {
+                if (child.name().equals("metadata")) {
+                    if (child.node().isEmpty()) {
+                        continue;
+                    }
+                    MetadataNode metaD = (MetadataNode) child.node().get();
+                    NodeList<AnnotationNode> annotations = metaD.annotations();
+                    for (AnnotationNode annotation : annotations) {
```

Comment:
This can be simplified as
```
for (AnnotationNode annotation : ((TypeDefinitionNode) moduleNode).metadata().get().annotations()) {
```

Reply:
Updated in [d6a53d3](https://github.com/ballerina-platform/persist-tools/pull/15/commits/d6a53d3f948c8f076acb3cc1f2730258a998cf40)